### ISSUE TITLE
Weight and sample loss don't change when I train

### ISSUE THREAD

**Issue by xizaoqu** [GlassyWing/stable-net#1]
Could you please upload your checkpoints and training params, thx.

**Comment by GlassyWing** [comment]
Actually，I face the same issue, the training is unstable even the net is called "stable". You can try the train_va.py which is much more stable and achieve equivalent effect.

**Comment by xizaoqu** [comment]
> Actually，I face the same issue, the training is unstable even the net is called "stable". You can try the train_va.py which is much more stable and achieve equivalent effect.

Thanks for replying.It seems that train_va.py doesn't use the weighted sample strategy, why can it achieve equivalent effect. Another question, when training with train.py using larger learning rate,the weights of all samples tend to decrease equally, and it seems that  smaller  weights always get smaller sample loss.If we set weights equal to zero,the loss is also zero,is it reasonable?

**Comment by GlassyWing** [comment]
> > Actually，I face the same issue, the training is unstable even the net is called "stable". You can try the train_va.py which is much more stable and achieve equivalent effect.
> 
> Thanks for replying.It seems that train_va.py doesn't use the weighted sample strategy, why can it achieve equivalent effect. Another question, when training with train.py using larger learning rate,the weights of all samples tend to decrease equally, and it seems that smaller weights always get smaller sample loss.If we set weights equal to zero,the loss is also zero,is it reasonable?

1. This method comes from the information bottleneck theory, but it is only information compression.
2. There is always a zero solution, so the paper requires the sum of all weights to be n, but this constraint is difficult to achieve in gradient descent (gradient descent prefers zero solutions), which is also the reason for the unstable training. Since the source code of the paper is not released, the method of imposing constraints is unknown. If you have any good suggestions, you are welcome to put forward them.

**Comment by xizaoqu** [comment]
> > > Actually，I face the same issue, the training is unstable even the net is called "stable". You can try the train_va.py which is much more stable and achieve equivalent effect.
> > 
> > 
> > Thanks for replying.It seems that train_va.py doesn't use the weighted sample strategy, why can it achieve equivalent effect. Another question, when training with train.py using larger learning rate,the weights of all samples tend to decrease equally, and it seems that smaller weights always get smaller sample loss.If we set weights equal to zero,the loss is also zero,is it reasonable?
> 
> 1. This method comes from the information bottleneck theory, but it is only information compression.
> 2. There is always a zero solution, so the paper requires the sum of all weights to be n, but this constraint is difficult to achieve in gradient descent (gradient descent prefers zero solutions), which is also the reason for the unstable training. Since the source code of the paper is not released, the method of imposing constraints is unknown. If you have any good suggestions, you are welcome to put forward them.

**Comment by xizaoqu** [comment]
> > > Actually，I face the same issue, the training is unstable even the net is called "stable". You can try the train_va.py which is much more stable and achieve equivalent effect.
> > 
> > 
> > Thanks for replying.It seems that train_va.py doesn't use the weighted sample strategy, why can it achieve equivalent effect. Another question, when training with train.py using larger learning rate,the weights of all samples tend to decrease equally, and it seems that smaller weights always get smaller sample loss.If we set weights equal to zero,the loss is also zero,is it reasonable?
> 
> 1. This method comes from the information bottleneck theory, but it is only information compression.
> 2. There is always a zero solution, so the paper requires the sum of all weights to be n, but this constraint is difficult to achieve in gradient descent (gradient descent prefers zero solutions), which is also the reason for the unstable training. Since the source code of the paper is not released, the method of imposing constraints is unknown. If you have any good suggestions, you are welcome to put forward them.

Thanks a lot, I'll learn more and try to figure it out.
